File: src/python/text-stream/claude.py
import anthropic

client = anthropic.Anthropic()

# モデルの選択
# claude-3-opus-20240229, claude-3-sonnet-20240229, claude-3-haiku-20240307

model = "claude-3-haiku-20240307"

with client.messages.stream(
    model=model,
    max_tokens=1000,  # 出力上限
    temperature=0.0,  # 0.0-1.0
    system="",  # 必要ならシステムプロンプトを設定
    messages=[
        {
            "role": "user",
            "content": "こんにちは、芝浦工業大学ShibaLabについて教えてください。",
        }
    ],
) as stream:
    for chunk in stream.text_stream:
        print(chunk, end="")

print("")

File: sample/javascript/text-stream/claude.js
import Anthropic from "@anthropic-ai/sdk";

const anthropic = new Anthropic({
  apiKey: process.env.ANTHROPIC_API_KEY,
});

async function main() {
  const stream = await anthropic.messages
    .stream({
      max_tokens: 1024,
      messages: [
        {
          role: "user",
          content: "こんにちは、芝浦工業大学ShibaLabについて教えてください。",
        },
      ],
      model: "claude-3-haiku-20240307",
    })
    .on("text", (text) => {
      console.log(text);
    });

  const message = await stream.finalMessage();
  console.log(message);
}

main();
